pkg/registry/seedmanagement/managedseed/strategy.go

```diff
@@ -65,12 +67,31 @@ func (s Strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object)
 	oldManagedSeed := old.(*seedmanagement.ManagedSeed)
 	newManagedSeed.Status = oldManagedSeed.Status
 
-	if !apiequality.Semantic.DeepEqual(oldManagedSeed.Spec, newManagedSeed.Spec) ||
-		oldManagedSeed.DeletionTimestamp == nil && newManagedSeed.DeletionTimestamp != nil {
+	if mustIncreaseGeneration(oldManagedSeed, newManagedSeed) {
 		newManagedSeed.Generation = oldManagedSeed.Generation + 1
 	}
 }
 
+func mustIncreaseGeneration(oldManagedSeed, newManagedSeed *seedmanagement.ManagedSeed) bool {
+	// The spec changed
+	if !apiequality.Semantic.DeepEqual(oldManagedSeed.Spec, newManagedSeed.Spec) {
+		return true
+	}
+
+	// The deletion timestamp was set
+	if oldManagedSeed.DeletionTimestamp == nil && newManagedSeed.DeletionTimestamp != nil {
+		return true
+	}
+
+	// The operation annotation was added with value "reconcile"
```

Comment:
Discussion regarding style:
For me comments are always a burden as they are in danger to be outdated when the code below changes but the comment was forgotten to be changed. In this case I think the function names already provide enough meaning so the comments are unnecessary. WDYT?
(this comment should not hold merging the PR back)